### PR TITLE
DAOS-11151 build: Update pmdk to 1.12.1~rc1

### PR DESCRIPTION
**Pull request by jolivier23**
Fixes issue with reserve API that can cause random crashes and
corruption in pool usage statistics.

PR-repos: pmdk@PR-24
Allow-unstable-test: true
Test-tag: pr daily_regression

Required-githooks: true

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -166,12 +166,10 @@ Requires: ndctl
 # needed to set PMem configuration goals in BIOS through control-plane
 %if (0%{?suse_version} >= 1500)
 Requires: ipmctl >= 02.00.00.3733
-# When 1.11.2 is released, we can change this to >= 1.11.2
-Requires: libpmemobj1 = 1.11.0-3.suse1500
+Requires: libpmemobj1 >= 1.12.1~rc1-1.suse1500
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:169]: Didn't we run into an issue in prior iterations of this update where we discovered that we were only setting a version on this requires for this one library but that this RPM also Requires others from pmdk that we should be setting the version on?

Comment (jolivier23) [reply utils/rpms/daos.spec:169]: so you think we should be adding libpmem and libpmemblk ?

Comment (jolivier23) [reply utils/rpms/daos.spec:169]: Is there any way to tell it to use libpmemobj1 >= [version] and ensure that libpmem and libpmemblk have same version ?

Comment (jolivier23) [reply utils/rpms/daos.spec:169]: though I assume it's a little different here since we are using >= and I presume we wouldn't get mixed versions again

Comment (brianjmurrell) [reply utils/rpms/daos.spec:169]: Ah, yes.  I suppose perhaps that issue previously was just with having the one pinned and not the rest.  So indeed, perhaps we can just rely on pinning one of them and the others will be the same version.  Although, quite honestly, I'd just rather remove the constraint altogether and rely on proper repo contents to get us what we want.  But I understand some people like the constraint.  Even if an improper repo configuration will cause the install to break anyway.

**Review by brianjmurrell — APPROVED** [review]

**Comment by jolivier23** [comment]
I ran daily_regression
All failures are seen in master branch:
DAOS-11282
DAOS-11377
DAOS-11380
DAOS-10818

**Pull request merged**